include/oneapi/dpl/execution

```diff
@@ -55,6 +55,10 @@
 #    include "oneapi/dpl/pstl/glue_numeric_ranges_impl.h"
 #endif // _ONEDPL_NUMERIC_RANGES_FORWARD_DECLARED
 
+#if _ONEDPL_ASYNC_FORWARD_DECLARED
```

Comment:
I think execution should not contain async API related stuff, because async API is a kind of a option and is enabled by explicit including <async> header. (the <range> has the same "option" logic)

Reply:
As I can see, the logic here is exactly the same as for other headers with the algorithms - inclusion of `<execution>` adds implementation if the declarations are already set by including the main header.

Do you mean that for `<async>` we should not at all use this two-stage logic with separate declarations and definitions, or what?

Reply:
Sorry, I missed that we already have the same logic with "optional" functionality "ranges".
And yes, while async is not included explicitly, `if _ONEDPL_ASYNC_FORWARD_DECLARED ` is not defined and we don't "pull" async stuff.
OK.